compiler/rustc_target/src/spec/wasm32_wasi_threads.rs

```diff
@@ -0,0 +1,58 @@
+use super::crt_objects::{self, LinkSelfContainedDefault};
+use super::{wasm_base, Cc, LinkerFlavor, Target};
+
+pub fn target() -> Target {
+    let mut options = wasm_base::options();
+
+    options.os = "wasi".into();
+    // options.add_pre_link_args(LinkerFlavor::WasmLld(Cc::Yes), &["--target=wasm32-wasi-threads"]);
+    options
+        .add_pre_link_args(LinkerFlavor::WasmLld(Cc::No), &["--import-memory", "--shared-memory"]);
```

Comment:
Should this perhaps also have the `--export-memory` argument as below too?

Reply:
I hoped to add it, but rust-lld doesn't support such an option
 ```= note: rust-lld: error: unknown argument: --export-memory```

Just checked If I just remove this line nothing changes. Guessed it gets overwritten by the next line, but if I reorder still nothing changes. Perhaps it's just a useless line

Reply:
If `Cc::Yes` has `--export-memory` below, then this should have it as well I think. If in testing it doesn't work then that may mean that the LLD you were using doesn't support `--export-memory` because it's too old.

This line should not be removed, however. It's required for the default mode of compiling the wasi target which is to use `rust-lld` without a c-compiler driver (`Cc::No`). It "works" if you remove this but then the output may not have shared memory since this argument won't be passed.

I'd recommend testing with a variety of programs to ensure the right "shape" of wasm module is produced. I wouldn't recommend exculusively using a small hello-world since that may not exercise what's necessary to test whether these options are working.

Reply:
> because it's too old.

Just tried with one from nightly and failed the same, maybe it's too old too. Do you know how to compile a one from the source code?

Reply:
✅